### DISCUSSION TITLE
Guide to create own widgets or extend existing?

### DISCUSSION THREAD

**Discussion by siddie** [h2oai/wave#2203] [Q&A] [answered]
Thank you for sharing this fantastic tool. I am looking to extend existing widgets and create my own. Can you direct me to the documentation on how I can start that journey?

**Comment by mturoci** [comment]
Hi @siddie!

There is currently no native support [yet](https://github.com/h2oai/wave/issues/35), but if you tell me what kind of custom widgets you need I can guide you further.

**Reply by siddie** [reply]
@mturoci 
Thank you for the swift response. As for the one I want to extend the most, it is Wide/Tall Series Stat.
![image](https://github.com/h2oai/wave/assets/1257332/8fbe6300-c1aa-4bab-912e-26e7876d7963)

What are the main pain points from the product perspective:
1. Those types of widgets are very often used to display a measured value delta. I can see that the 'aux_value' field is best suited for that. But the essential thing missing from UX perspective is being able to display a positive delta in green color, negative - in red. So, if I could change the color of aux value - that would dramatically improve the UX.
2. I want to add "I(nfo)" sub-element, so that when clicking or hovering I can display an explanation on what and how exactly is being measured.
3. I want the Wide/Tall Series Stat widget tile clickable so that on click I can display a more detailed view.

**Reply by mturoci** [reply, marked as answer]
1 a 3 can be achieved by [custom JS](https://wave.h2o.ai/docs/javascript/). 2 can be achieved with JS as well, but I would probably recommend putting all the extra info into a dedicated page since I assume there is going to be more than 1 widget to explain and the explanation can be quite lengthy so using tooltip would be inferior UX compared to regular text.

**Reply by siddie** [reply]
Thanks @mturoci . I'll give a try to custom JS
